### PR TITLE
Update to example apps to support React 17 testing

### PR DESCRIPTION
**Pull request by LFDanLu**
Closes 

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 🧢 Your Project:

RSP

### CODE REVIEW

File: examples/rac-tailwind/src/index.js [hunk 1]
```diff
@@ -3,3 +3,10 @@ import { App } from './App';
 
 let root = createRoot(document.getElementById('root'));
 root.render(<App />);
+
+// TODO: check on react-dom's version didn't work, figure out later
+// The below is for when react 17 is installed
+// import ReactDOM from 'react-dom';
+// ReactDOM.render(
+//   <App />, document.getElementById("root")
+// )
```
Comment (LFDanLu) [inline examples/rac-tailwind/src/index.js:12]: Was doing a ReactDOM version check like in the other apps to decide which ReactDOM render call to use but it broke here in the tailwind app. Manually checked that our latest code on main worked in react 17 here, will followup on making this check more automatic

File: examples/rsp-next-ts-17/.eslintrc [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (LFDanLu) [inline examples/rsp-next-ts-17/.eslintrc:1]: The bulk of this rsp next + react 17 example app is just copy pasta'd from our existing one

File: examples/rsp-next-ts-17/package.json [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  "homepage": ".",
+  "scripts": {
+    "dev": "next dev",
+    "build": "next build",
+    "export": "next export",
+    "start": "next start",
+    "lint": "next lint"
+  },
+  "dependencies": {
+    "@adobe/react-spectrum": "^3.26.0",
```
Comment (LFDanLu) [inline examples/rsp-next-ts-17/package.json:14]: Verified that this example app builds properly with latest RSP, but using the latest nightly and/or local verdaccio blew up in the build with errors like
```
./node_modules/@internationalized/date/dist/import.mjs
Module not found: Can't resolve '@swc/helpers/_/_class_private_field_init' in '...react-spectrum/examples/rsp-next-ts-17/node_modules/@internationalized/date/dist'

Import trace for requested module:
./node_modules/@internationalized/date/dist/import.mjs
./node_modules/@react-spectrum/calendar/dist/import.mjs
./node_modules/@adobe/react-spectrum/dist/import.mjs
```
Not exactly sure why, but confirmed that the new useId stuff works in a separate example app using react 17, next 11, and webpack 4 via https://github.com/devongovett/rsp-next

**Review by snowystinger — COMMENTED** [review]
Looks pretty good, would you mind making a commit with the circle ci build stuff turned on for this branch and then another to set it back once we have the build comments?

since I've been bit waiting to see what happens on main before doing something similar

**Review by snowystinger — COMMENTED** [review]
Would be good for the links generated to have different names in the comment instead of two called `[NextJS Test App](...)`

Looks like most of the didn't build for some reason? or the links aren't good for some reason?

**Comment by LFDanLu** [comment]
ah, the react 17 variants aren't being built and added as links via CI yet, that is to come as follow up. As for the dead links, its a quirk of running the build on a branch, it comments early before the verdaccio stuff finishes since we have two "comment" calls in the circleci process (one for PRs that is standard but now takes a different code due to the change in comment.js and one after verdaccio finishes). The links should work now

**Review by snowystinger — APPROVED** [review]

File: examples/rsp-webpack-4/src/index.js [hunk 4]
```diff
@@ -1,11 +1,17 @@
...
 import './index.css';
 import App from './App';
 
-const root = ReactDOM.createRoot(
-  document.getElementById('root')
-);
-root.render(
-  <App />
-);
+if (ReactDOM.version.startsWith('18')) {
```
Comment (LFDanLu) [inline examples/rsp-webpack-4/src/index.js:5]: Will follow up and add something to the webpack 4 example app so it actually render a page we can test. For now I've confirmed in a separate webpack 4 app that the latest nightly with the useId changes works with React 17/18

**Review by ktabors — APPROVED** [review]

**Pull request merged**